In [1]:
pip install ipython-sql mysql-connector-python

Note: you may need to restart the kernel to use updated packages.


In [3]:
%load_ext sql

In [5]:
%sql mysql+mysqlconnector://root:datapoet@localhost/dap1

#### 2.4.1 **World Bank: Indicators**

In the data exploration phase, several **potentially relevant indicators** from the World Bank were selected for analysis. The initial focus was on constructing visualizations in **Power BI** to assess which indicators were sufficiently consistent to include in the exploratory analysis.

The four indicators are:

- GDP per capita (current US$)
- Net migration
- Poverty headcount ratio at national poverty lines (% of population)
- Unemployment, total (% of total labor force) (modeled ILO estimate)

Before importing the data into Power BI, the table **wb_values** was unpivoted using the query:



⬇️ **wb_values_v_unpivoted**

In [9]:
%%sql
SELECT * FROM wb_values_v_unpivoted LIMIT 200;

 * mysql+mysqlconnector://root:***@localhost/dap1
200 rows affected.


[(1, 'GDP per capita (current US$)', 1, 'Aruba', '1960', None),
 (1, 'GDP per capita (current US$)', 2, 'Afghanistan', '1960', None),
 (1, 'GDP per capita (current US$)', 3, 'Angola', '1960', None),
 (1, 'GDP per capita (current US$)', 4, 'Albania', '1960', None),
 (1, 'GDP per capita (current US$)', 5, 'Andorra', '1960', None),
 (1, 'GDP per capita (current US$)', 6, 'United Arab Emirates', '1960', None),
 (1, 'GDP per capita (current US$)', 7, 'Argentina', '1960', None),
 (1, 'GDP per capita (current US$)', 8, 'Armenia', '1960', None),
 (1, 'GDP per capita (current US$)', 9, 'American Samoa', '1960', None),
 (1, 'GDP per capita (current US$)', 10, 'Antigua and Barbuda', '1960', None),
 (1, 'GDP per capita (current US$)', 11, 'Australia', '1960', 1810.59744278609),
 (1, 'GDP per capita (current US$)', 12, 'Austria', '1960', 943.610800103121),
 (1, 'GDP per capita (current US$)', 13, 'Azerbaijan', '1960', None),
 (1, 'GDP per capita (current US$)', 14, 'Burundi', '1960', 71.3602242458753),
 (1, 'GDP per capita (current US$)', 15, 'Belgium', '1960', 1290.28607216192),
 (1, 'GDP per capita (current US$)', 16, 'Benin', '1960', 90.0358313130591),
 (1, 'GDP per capita (current US$)', 17, 'Burkina Faso', '1960', 69.0831953321335),
 (1, 'GDP per capita (current US$)', 18, 'Bangladesh', '1960', 84.8253371946916),
 (1, 'GDP per capita (current US$)', 19, 'Bulgaria', '1960', None),
 (1, 'GDP per capita (current US$)', 20, 'Bahrain', '1960', None),
 (1, 'GDP per capita (current US$)', 21, 'Bahamas, The', '1960', 1483.00368182207),
 (1, 'GDP per capita (current US$)', 22, 'Bosnia and Herzegovina', '1960', None),
 (1, 'GDP per capita (current US$)', 23, 'Belarus', '1960', None),
 (1, 'GDP per capita (current US$)', 24, 'Belize', '1960', 307.12862895614),
 (1, 'GDP per capita (current US$)', 25, 'Bermuda', '1960', 1902.40208533539),
 (1, 'GDP per capita (current US$)', 26, 'Bolivia', '1960', 100.673901521639),
 (1, 'GDP per capita (current US$)', 27, 'Brazil', '1960', None),
 (1, 'GDP per capita (current US$)', 28, 'Barbados', '1960', 344.103978900915),
 (1, 'GDP per capita (current US$)', 29, 'Brunei Darussalam', '1960', None),
 (1, 'GDP per capita (current US$)', 30, 'Bhutan', '1960', None),
 (1, 'GDP per capita (current US$)', 31, 'Botswana', '1960', 59.2971126032832),
 (1, 'GDP per capita (current US$)', 32, 'Central African Republic', '1960', 66.7700952131282),
 (1, 'GDP per capita (current US$)', 33, 'Canada', '1960', 2264.94849658191),
 (1, 'GDP per capita (current US$)', 34, 'Switzerland', '1960', 1954.3113206427),
 (1, 'GDP per capita (current US$)', 35, 'Channel Islands', '1960', None),
 (1, 'GDP per capita (current US$)', 36, 'Chile', '1960', 504.801137829134),
 (1, 'GDP per capita (current US$)', 37, 'China', '1960', 89.5202142368479),
 (1, 'GDP per capita (current US$)', 38, "Côte d'Ivoire", '1960', 147.277834060423),
 (1, 'GDP per capita (current US$)', 39, 'Cameroon', '1960', 120.018193753266),
 (1, 'GDP per capita (current US$)', 40, 'Congo, Dem. Rep.', '1960', None),
 (1, 'GDP per capita (current US$)', 41, 'Congo, Rep.', '1960', 124.782358870445),
 (1, 'GDP per capita (current US$)', 42, 'Colombia', '1960', 256.962847338565),
 (1, 'GDP per capita (current US$)', 43, 'Comoros', '1960', None),
 (1, 'GDP per capita (current US$)', 44, 'Cabo Verde', '1960', None),
 (1, 'GDP per capita (current US$)', 45, 'Costa Rica', '1960', 376.968785602974),
 (1, 'GDP per capita (current US$)', 46, 'Cuba', '1960', None),
 (1, 'GDP per capita (current US$)', 47, 'Curaçao', '1960', None),
 (1, 'GDP per capita (current US$)', 48, 'Cayman Islands', '1960', None),
 (1, 'GDP per capita (current US$)', 49, 'Cyprus', '1960', None),
 (1, 'GDP per capita (current US$)', 50, 'Czechia', '1960', None),
 (1, 'GDP per capita (current US$)', 51, 'Germany', '1960', 1157.88046394509),
 (1, 'GDP per capita (current US$)', 52, 'Djibouti', '1960', None),
 (1, 'GDP per capita (current US$)', 53, 'Dominica', '1960', None),
 (1, 'GDP per capita (current US$)', 54,

The **wb_values_v_unpivoted** view was better suited for manipulation and analysis in Power BI.

Upon reviewing the visualizations, the **poverty headcount ratio** indicator was excluded from further analysis due to the lack of consistent data across countries. While relevant for analyzing national poverty, the inconsistency in available data limited its comparability between countries.

Subsequently, trend lines were examined across the different indicators, followed by a study of correlations between countries and indicators to derive potentially insightful relationships.

#### Correlation between **Argentina's net migration** and **Argentina's unemployment rate**
World Bank Indicators/ 1991-2023

In [24]:
%%sql
SELECT
    (
        (COUNT(*) * SUM(arg_mig.value * arg_unemp.value)) - (SUM(arg_mig.value) * SUM(arg_unemp.value))
    ) / 
    SQRT(
        (COUNT(*) * SUM(POW(arg_mig.value, 2)) - POW(SUM(arg_mig.value), 2)) *
        (COUNT(*) * SUM(POW(arg_unemp.value, 2)) - POW(SUM(arg_unemp.value), 2))
    ) AS correlation
FROM
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Net migration' AND year > 1990) arg_mig
JOIN
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) arg_unemp
ON arg_mig.year = arg_unemp.year;

 * mysql+mysqlconnector://root:***@localhost/dap1
1 rows affected.


[(-0.7212184223819235,)]

#### Correlation between **Argentina's net migration** and **Spain's unemployment rate**
World Bank Indicators/ 1991-2023

In [32]:
%%sql
SELECT
    (
        (COUNT(*) * SUM(arg_mig.value * sp_unemp.value)) - (SUM(arg_mig.value) * SUM(sp_unemp.value))
    ) / 
    SQRT(
        (COUNT(*) * SUM(POW(arg_mig.value, 2)) - POW(SUM(arg_mig.value), 2)) *
        (COUNT(*) * SUM(POW(sp_unemp.value, 2)) - POW(SUM(sp_unemp.value), 2))
    ) AS correlation
FROM
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Net migration' AND year > 1990) arg_mig
JOIN
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Spain' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) sp_unemp
ON arg_mig.year = sp_unemp.year;

 * mysql+mysqlconnector://root:***@localhost/dap1
1 rows affected.


[(0.3423962431300349,)]

#### Correlation between **Argentina's net migration** and **Spain's unemployment rate**
World Bank Indicators/ 1991-2023

In [35]:
%%sql
SELECT
    (
        (COUNT(*) * SUM(arg_mig.value * it_unemp.value)) - (SUM(arg_mig.value) * SUM(it_unemp.value))
    ) / 
    SQRT(
        (COUNT(*) * SUM(POW(arg_mig.value, 2)) - POW(SUM(arg_mig.value), 2)) *
        (COUNT(*) * SUM(POW(it_unemp.value, 2)) - POW(SUM(it_unemp.value), 2))
    ) AS correlation
FROM
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Net migration' AND year > 1990) arg_mig
JOIN
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Italy' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) it_unemp
ON arg_mig.year = it_unemp.year;


 * mysql+mysqlconnector://root:***@localhost/dap1
1 rows affected.


[(0.15511478083269578,)]

#### Correlation between **Spain's unemployment rate** and **Italy's unemployment rate**
World Bank Indicators/ 1991-2023

In [37]:
%%sql
SELECT
    (
        (COUNT(*) * SUM(arg_unemp.value * sp_unemp.value)) - (SUM(arg_unemp.value) * SUM(sp_unemp.value))
    ) / 
    SQRT(
        (COUNT(*) * SUM(POW(arg_unemp.value, 2)) - POW(SUM(arg_unemp.value), 2)) *
        (COUNT(*) * SUM(POW(sp_unemp.value, 2)) - POW(SUM(sp_unemp.value), 2))
    ) AS correlation
FROM
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) arg_unemp
JOIN
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Spain' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) sp_unemp
ON arg_unemp.year = sp_unemp.year;


 * mysql+mysqlconnector://root:***@localhost/dap1
1 rows affected.


[(-0.16260482826243824,)]

#### Correlation between **Argentina's unemployment rate** and **Italy's unemployment rate**
World Bank Indicators/ 1991-2023

In [41]:
%%sql
SELECT
    (
        (COUNT(*) * SUM(arg_unemp.value * it_unemp.value)) - (SUM(arg_unemp.value) * SUM(it_unemp.value))
    ) / 
    SQRT(
        (COUNT(*) * SUM(POW(arg_unemp.value, 2)) - POW(SUM(arg_unemp.value), 2)) *
        (COUNT(*) * SUM(POW(it_unemp.value, 2)) - POW(SUM(it_unemp.value), 2))
    ) AS correlation
FROM
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Argentina' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) arg_unemp
JOIN
    (SELECT year, value FROM wb_values_v_unpivoted WHERE country_name = 'Italy' AND indicator_name = 'Unemployment, total (% of total labor force) (modeled ILO estimate)' AND year > 1990) it_unemp
ON arg_unemp.year = it_unemp.year;


 * mysql+mysqlconnector://root:***@localhost/dap1
1 rows affected.


[(0.2065892633840354,)]

The correlation values between the selected indicators were generally weak, ranging between 0 and 0.3, indicating a **lack of strong relationships**. One notable exception was the correlation between Argentina's Net Migration and Argentina's Unemployment, which showed a value of -0.72, though this still **does not reflect a robust correlation**. These results suggest that these variables are not directly related within the scope of our dataset, aligning with the understanding that such phenomena are typically influenced by multiple factors.